### PR TITLE
feat: add drop yield

### LINKED ISSUE #281: Add CFG and DROP APY
https://github.com/centrifuge/apps/issues/281

## Background
We wanna show the CFG yield as well as the actual 30d DROP yields now.

## How to derive CFG yields
Prerequisite is a price feed from UniSwap, technical ticket/specs to be drafted.
Based on the price of CFG in USD the yield can be approximated as:
$$
CFG yield = daily rewards rate * days per year * CFG token price
$$
with number of days = 365 and the token price in decimal format, e.g. 10 CFG per day per 10k investment --> 0.001
so e.g. assuming a token price of 0.7
$$
CFG yield = 0.001 * 365 * 0.7 = 0.2555 = 25.55 pc
$$
An example calculation for understanding the economics can be found [here](https://docs.google.com/spreadsheets/d/1MC7DtfUk8Ztc0MaRfCc3wt2OF7q0YbJ8nbRF3-_-5ds/edit#gid=2014695545)

### Dashboard

image
[Figma](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5861%3A21590)

#### Update introduction paragraph 
- Expandable detail: `The Tinlake protocol is rewarding every Tinlake investment daily in CFG token, the Centrifuge Chain’s native token. The Investment Reward Rate (APR)” is an annualized representation of these rewards considering the current CFG token market price taken from CoinGecko and current daily Tinlake protocol reward rate (see related governance discussions and decisions here). Note that these are rewards from the protocol for providing liquidity to the ecosystem independent from the pool, it’s issuers, their asset originators, or any Centrifuge entity. There is no guarantee that an investor will receive those rewards nor that the current annualized reward APR will be met. This is not investment advice — please see the Investment Disclaimer for more info.`
- Secondary link: [Get started](https://developer.centrifuge.io/learn/understanding-tinlake/) open in new window

#### (1) CFG Yield
- The value should be calculated/updated on every page load
- The  calculated value of the `CFG yield rate` is shown on the Tinlake dashboard in the second summary card on top of the pools (note the changed card design)
- Label `Reward Rate (APR)`

#### (2) DROP yields - 30d
- In the pool list items, the `DROP APR` is replaced by the `DROP APY` (duration of 30 days). 
- The DROP APR is shown as a subline to the DROP APY

#### (3) TIN yields - 3m
Please don't show yet, but leave behind the feature flag as it currently is

#### (4) Tooltips
- Tooltip on CFG yield in header
    - `The annualized CFG reward rate for investments, based on the current CFG token market price and the daily reward rate` 
Note that these are rewards from the protocol for providing liquidity to the ecosystem independent from the pool, it’s issuers, their asset originators, or any Centrifuge entity. There is no guarantee that an investor will receive those rewards nor that the current annualized reward APR will be met. This is not investment advice - please see the [Investment disclaimer](open investment disclaimer on click) for more info.`
- Tooltip on 30d DROP Yield column header
    - `The 30d DROP yield is the effective annualized return of the pool's DROP token over the last 30 days.`
- Tooltip on`x.00 % APR` in pool component
    - `DROP tokens earn yield on the outstanding assets at the DROP APR. The effective APY may deviate due to compounding effects or unused liquidity in the pool reserve.`
    
#### (1) In header

[Figma: DROP yield in summary of pool overview](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5567%3A14612)
- Show 30d yields
- show APR below
- Update Tooltip
    - `DROP tokens earn yield on the outstanding assets at the DROP APR. The effective annual yield may deviate due to compounding effects or unused liquidity in the pool reserve. The 30d DROP APY is the effective annualized return of the pool's DROP token over the last 30 days.`

#### (2) In pool balance display

- Replace APR by yield in DROP component
    - `<value with one digit> % APY (30 days)`: 

### View: Investment component

- Add `DROP APY (30 days): <value>`
- Add Tooltip over DROP APR and yield: 
        - `DROP tokens earn yield on the outstanding assets at the DROP APR. The effective APY may deviate due to compounding effects or unused liquidity in the pool reserve. The 30d DROP APY is the effective annualized return of the pool's DROP token over the last 30 days.`

Note, we will try to include yield calculations in the documentation. If we manage to do that before implementation, we'll add links to this in the tool tips.

#### Update investment disclaimer (Overlay)
```
Nothing contained in this website is to be construed as a solicitation or offer, or recommendation, to buy or sell any interest in any note or other security, or to engage in any other transaction, and the content herein does not constitute, and should not be considered to constitute, an offer of securities. No statement herein made constitutes an offer to sell or a solicitation of an offer to buy a note or other security. All information on this Web page is provided and maintained by the issuers of the respective Tinlake pools. The issuers have full responsibility. Please contact the respective issuer in case of any inquiries. Centrifuge and its affiliates are not liable nor responsible for the information provided hereby.

Before investing in any of the Tinlake pools, please check the issuer's offering materials and subscription documents including the Executive Summary to understand the terms, conditions, and investment risks of each pool. The issuer provides investment risk factors which are important to understand when you consider whether to invest in a pool. You alone assume the sole responsibility of evaluating the merits and risks associated with the use of any information or other content before making any decisions based on such information or other content.

The Tinlake protocol is rewarding every Tinlake investment daily in CFG token, the Centrifuge Chain’s native token. Note that these are rewards from the protocol for providing liquidity to the ecosystem independent from the pool, it’s issuers, their asset originators, or any Centrifuge entity. There is no guarantee that an investor will receive those rewards.
```

#### Addendum: DROP APY in upcoming pools and pools live less than 30 days
If a pool has not yet been live for more than 30 days, or is not live yet (i.e. upcoming), the value of DROP APY is not known. Until the DROP APY is known, the system shows the APR as a "good enough estimation", using the label `Expected APR: <value of estimated APR>` in the context where it applies (i.e. pool item on dashboard, pool overview, pool balance).

##### Pool list item (dashboard)
- Label `Expected APR`

image[Figma](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5848%3A19895)

##### Pool overview
Summary:
- Label: `Expected DROP APR`

image
[Figma](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5848%3A19804)

Pool Balance:
- Label `Expected APR`

image
[Figma](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5848%3A20028)

#### Investment capacity
- No APY
- `Expected DROP APR`

image
[Figma](https://www.figma.com/file/EWziKt1qCFdHGhWdnkrcKf/Tinlake-UI?node-id=5853%3A19477)

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

Closes #281 

Use this preview link since it uses `mainnet` contracts:  https://610490a686f404215babd15a--tinlake-mainnet-production.netlify.app

This pull request makes some UI and data changes to `tinlake-ui`:
- added `DROP` 30d APY to the pool list
- added `TIN` 3m APY to the pool list (behind `?showAll=true` feature flag)
- added `DROP` 30d APY to the investment overview of a pool's overview page
- added `DROP` 30d APY to the tranche overview of a pool's overview page
- added `DROP` 30d APY to the `Invest` component of a pool's investment page
- updated various verbiage
- added/updated several tooltips
  - Note: I could not add a tooltip to the values within the pool list. The tooltip component was behaving unexpectedly. Perhaps we could revisit this feature in the future.

### CODE REVIEW

File: tinlake-ui/utils/hooks.ts [hunk 1]
```diff
@@ -18,3 +23,41 @@ export const useInterval = (callback: any, delay: number) => {
     }
   }, [delay])
 }
+
+export const useYield = () => {
```
Comment (onnovisser) [inline tinlake-ui/utils/hooks.ts]: 🙌

File: tinlake-ui/utils/hooks.ts [hunk 2]
```diff
@@ -18,3 +23,41 @@ export const useInterval = (callback: any, delay: number) => {
...
+  const pool = useSelector<any, PoolState>((state) => state.pool)
+  const pools = useSelector<any, PoolsState>((state) => state.pools)
+
+  const [dropYield, setDropYield] = React.useState('')
+  const [tinYield, setTinYield] = React.useState('')
+
+  React.useEffect(() => {
+    if (pools.data?.pools && pool.poolId) {
```
Comment (onnovisser) [inline tinlake-ui/utils/hooks.ts:35]: I think `useMemo` is better suited here. Removes the need for the `useState` s

File: tinlake-ui/utils/hooks.ts [hunk 3]
```diff
@@ -18,3 +23,41 @@ export const useInterval = (callback: any, delay: number) => {
...
+  return { dropYield, tinYield }
+}
+
+export const useCFGYield = () => {
+  const rewards = useSelector<any, RewardsState>((state: any) => state.rewards)
+  const wCFGPrice = useSelector<any, string>((state: any) => state.userRewards.wCFGPrice)
+
+  const [cfgYield, setCFGYield] = React.useState('0')
+
+  React.useEffect(() => {
```
Comment (onnovisser) [inline tinlake-ui/utils/hooks.ts:53]: Same here

Comment (onnovisser) [reply tinlake-ui/utils/hooks.ts:53]: Though I'm not sure the computation is even heavy enough to merit it.

**Review by onnovisser — COMMENTED** [review]
Nice! Left just a couple comments

File: tinlake-ui/utils/hooks.ts (hunk 3 above)
Comment (jpangelle) [reply tinlake-ui/utils/hooks.ts:56]: Nice!

File: tinlake-ui/utils/hooks.ts (hunk 2 above)
Comment (jpangelle) [reply tinlake-ui/utils/hooks.ts:33]: Refactored 🚀

**Comment by jpangelle** [comment]
@onnovisser Can you rereview?
@armandomistral Can you also review?

File: tinlake-ui/static/tooltips.json [hunk 4]
```diff
@@ -92,5 +92,17 @@
   "oversubscribed": {
     "title": "Oversubscribed pool",
     "description": "This pool is currently oversubscribed and not open for investments. Locked investment orders may be pending until the pool opens again."
+  },
+  "dropApy": {
+    "title": "Drop APY",
```
Comment (armandomistral) [inline tinlake-ui/static/tooltips.json:97]: @jpangelle, let's set "Drop" in uppercase, i.e. "DROP APY".

File: tinlake-ui/static/tooltips.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
   },
   "dropAPR": {
     "title": "DROP APR",
```
Comment (armandomistral) [inline tinlake-ui/static/tooltips.json:20]: @jpangelle, I'm pretty sure the title should be changed to "DROP APY" here.

File: tinlake-ui/static/tooltips.json [hunk 6]
```diff
@@ -92,5 +92,17 @@
   "oversubscribed": {
     "title": "Oversubscribed pool",
     "description": "This pool is currently oversubscribed and not open for investments. Locked investment orders may be pending until the pool opens again."
+  },
+  "dropApy": {
+    "title": "Drop APY",
+    "description": "The 30d DROP yield is the effective annualized return of the pool's DROP token over the last 30 days."
+  },
+  "dropApr": {
+    "title": "Drop APR",
```
Comment (armandomistral) [inline tinlake-ui/static/tooltips.json:101]: @jpangelle, let's consistently set token in capitals, i.e. "DROP APR"

File: tinlake-ui/static/tooltips.json [hunk 7]
```diff
@@ -92,5 +92,17 @@
...
+    "description": "DROP tokens earn yield on the outstanding assets at the DROP APR. The effective APY may deviate due to compounding effects or unused liquidity in the pool reserve."
+  },
+  "dropInvestment": {
+    "title": "Drop Investment",
```
Comment (armandomistral) [inline tinlake-ui/static/tooltips.json:105]: @jpangelle, I'd also change this one to "DROP Investment"

**Review by armandomistral — APPROVED** [review]
Hi, @jpangelle, just commented on some instances asking to capitalize the token label

**Review by onnovisser — APPROVED** [review]

**Pull request merged**